Meetup Client: Refactor to stop using v2 API endpoints

coreymckrill opened the pull request:
Meetup will apparently shut down all of the v2 API endpoints on 2019-08-15. Our client was still using the v2 `events` endpoint, so this refactor changes that to get as close as possible to the same dataset using v3 endpoints. It's not a perfect match, though, and the v3 API is not friendly towards our use case of pulling events from a large number of different groups all at once.

Client changes:
* Refactor the `get_events` method
* Remove unused signing method
* Add a request header for accepting json content type
* Adjust the throttle mechanism to be a bit more conservative
* Sanitize group slugs before using them in API endpoint requests

This also refactors the Meetup Events report in the WordCamp Reports plugin, since the refactored `get_events` method isn't a dropin replacement.

**To test**

Load this changeset into your wporg sandbox and then try running the [Meetup Events report](https://central.wordcamp.org/wp-admin/index.php?page=wordcamp-reports&report=meetup-events). It will take a really long time, but should produce results.

coreymckrill committed:
Remove unused request signing method

This method is no longer used now that we have the OAuth client

coreymckrill committed:
Meetup Client: Refactor the `get_events` method to for API v3

The Meetup API v2 events endpoint is being shut down, so this tries
to retrieve the same kind of data using v3 endpoints. Unfortunately
there is no equivalent v3 endpoint that can retrieve events for
multiple groups at once, so we have to request events from one. group.
at. a. time.

The inputs are not quite the same, and the response data from the
Meetup API is also slightly different, so the places we're using this
method in our code will also have to be refactored (this will be in
separate commits).

coreymckrill committed:
Meetup Client: Add request header for json content type

This is just a best practice and a precaution in case Meetup changes
responses from their API without notice. Which they have done before.

coreymckrill committed:
Meetup Client: Tweak the request throttle mechanism

This increases the request quota threshold at which we will pause to
avoid getting throttled. Since we will potentially be making hundreds
of requests in a row while retrieving events now, we need to be more
cautious about other processes using the API simultaneously causing
an accidental throttle.

coreymckrill committed:
Meetup Client: Sanitize group slugs for use in requests, and doc cleanup

coreymckrill committed:
Reports: Refactor Meetup Events to use refactored API client method

iandunn commented:
If you haven't already, it'd be good to compare the results for the same report queries before and after applying the branch, to ensure that they match exactly. I've caught unexpected errors w/ this type of change in the past that way, and if they're missed they could mess up stats, especially w/ year-to-year comparisons.

coreymckrill committed:
Reports: Increase memory and max execution time for Meetup Events

This is necessary to avoid timing out while sending a separate API
request for every single meetup group in the chapter program.

coreymckrill commented:
Good idea.

I used the Meetup Events report to compare total event numbers for 2017 and 2018 between the current, old endpoint and the new one on this branch. The numbers were a bit different, but that turned out to be due to a change in the report logic that is also made in this branch. After adjusting for that, the numbers are nearly identical (2018 is identical, 2017 is off by 8). I don't know how to account for that minor discrepancy, but it could be something within the logic of the v3 endpoint. I'm inclined not to worry about it.

coreymckrill committed:
Reports: Add another filter to groups in Meetup Events

The more groups that have to be queried, the longer this report takes.
This filters out groups whose last event was before the beginning of
the given date range. It's less effective the farther back in time the
date range is. But there are also less total groups the farther back
in time we go...

coreymckrill committed:
Reports: Add max date interval to Meetup Events report

Piggybacking on the last commit, this prevents the report from being
run with a date range that covers more than one year. That will limit
the total number of events that get retrieved and also help limit the
number of groups queried when the start of the date range is a long
ways back.
